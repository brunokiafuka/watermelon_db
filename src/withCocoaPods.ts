import {ExpoConfig} from "@expo/config-types";
import {withDangerousMod} from "@expo/config-plugins";
import path from "path";
import resolveFrom from "resolve-from";
import filesys from "fs";

const fs = filesys.promises;

function isWatermelonDBInstalled(projectRoot: string) {
    const resolved = resolveFrom.silent(
        projectRoot,
        "@nozbe/watermelondb/package.json"
    );
    return resolved ? path.dirname(resolved) : null;
}

export const withCocoaPods = (config: ExpoConfig): ExpoConfig => {
    return withDangerousMod(config, [
        "ios",
        async (config) => {
            const filePath = path.join(
                config.modRequest.platformProjectRoot,
                "Podfile"
            );

            const contents = await fs.readFile(filePath, "utf-8");

            const watermelonPath = isWatermelonDBInstalled(
                config.modRequest.projectRoot
            );

            if (watermelonPath) {
                await fs.writeFile(filePath, updatePodfile(contents));
            } else {
                throw new Error("Please make sure you have watermelondb installed");
            }
            return config;
        },
    ]) as ExpoConfig;
};

export function updatePodfile(contents: string): string {
    let newContent = contents;
    if (!contents.includes("pod 'simdjson'")) {
        const patchKey = "post_install";
        const slicedContent = contents.split(patchKey);
        slicedContent[0] += `\n
  pod 'simdjson', path: File.join(File.dirname(\`node --print "require.resolve('@nozbe/simdjson/package.json')"\`)), :modular_headers => true \n\n  `;
        newContent = slicedContent.join(patchKey);
    }

    return newContent
}
