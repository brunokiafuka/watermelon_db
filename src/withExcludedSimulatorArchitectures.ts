import {ExpoConfig} from "@expo/config-types";
import {withXcodeProject} from "@expo/config-plugins";

/**
 * Exclude building for arm64 on simulator devices in the pbxproj project.
 * Without this, production builds targeting simulators will fail.
 */
// @ts-ignore
export function setExcludedArchitectures(project) {

    const configurations = project.pbxXCBuildConfigurationSection();
    // @ts-ignore
    for (const {buildSettings} of Object.values(configurations || {})) {
        // Guessing that this is the best way to emulate Xcode.
        // Using `project.addToBuildSettings` modifies too many targets.
        if (
            typeof (buildSettings === null || buildSettings === void 0
                ? void 0
                : buildSettings.PRODUCT_NAME) !== "undefined"
        ) {
            buildSettings['"EXCLUDED_ARCHS[sdk=iphonesimulator*]"'] = '"arm64"';
        }
    }

    return project;
}

function logToFile(content: string, file = 'log.txt') {
    const fs = require('fs');
    fs.writeFileSync(file, content);
}

export const withExcludedSimulatorArchitectures = (c: ExpoConfig): ExpoConfig => {
    return withXcodeProject(c, (config) => {
        config.modResults = setExcludedArchitectures(config.modResults);
        return config;
    }) as ExpoConfig;
};
