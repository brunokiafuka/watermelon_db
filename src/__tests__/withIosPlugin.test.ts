import fs from 'fs/promises';
import path from 'path';
import {updatePodfile} from "../withCocoaPods";
import {setExcludedArchitectures} from "../withExcludedSimulatorArchitectures";

describe('iOS plugin', () => {
    let PodfileContents: string;

    // TODO: Can this be typed better?
    let XCodeProject: {
        pbxXCBuildConfigurationSection: () => any
    };
    // TODO: Can this be typed better?
    let XCodeProjectConfigurations: object;
    let XCodeProjectConfigurationsFixture: string;

    beforeAll(async () => {
        PodfileContents = await fs.readFile(
            path.resolve(__dirname, './fixtures/Podfile'),
            'utf-8');

        XCodeProjectConfigurationsFixture = await fs.readFile(
            // Contents of this json file were obtained from a vanilla expo project
            path.resolve(__dirname, './fixtures/xcodeproject.config.json'),
            'utf-8');

    })

    beforeEach(async () => {
        XCodeProjectConfigurations = JSON.parse(XCodeProjectConfigurationsFixture);
        XCodeProject = {
            pbxXCBuildConfigurationSection(){
                return XCodeProjectConfigurations;
            }
        }


    })

    it('should update the Podfile with simdjson', async () => {
        const updatedPodfile = updatePodfile(PodfileContents);
        expect(updatedPodfile).toMatchSnapshot();
    })

    it('should update the Podfile with simdjson only once on multiple prebuilds', async () => {
        let updatedPodfile = updatePodfile(PodfileContents);
        updatePodfile(PodfileContents)
        expect(updatedPodfile).toMatchSnapshot();
    })

    it('should be able to update the XCode project with excluded architectures', async () => {
        setExcludedArchitectures(XCodeProject);
        expect(XCodeProject.pbxXCBuildConfigurationSection()).toMatchSnapshot();
    })

    it('should be able to update the XCode project even if run twice', async () => {
        setExcludedArchitectures(XCodeProject);
        setExcludedArchitectures(XCodeProject);
        expect(XCodeProject.pbxXCBuildConfigurationSection()).toMatchSnapshot();
    })


})
