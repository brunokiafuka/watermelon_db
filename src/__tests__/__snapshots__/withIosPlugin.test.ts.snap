// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iOS plugin should be able to update the XCode project even if run twice 1`] = `
{
  "13B07F941A680F5B00A75B9A": {
    "baseConfigurationReference": "6C2E3173556A471DD304B334",
    "baseConfigurationReference_comment": "Pods-expovanillasdk50.debug.xcconfig",
    "buildSettings": {
      ""EXCLUDED_ARCHS[sdk=iphonesimulator*]"": ""arm64"",
      "ASSETCATALOG_COMPILER_APPICON_NAME": "AppIcon",
      "CLANG_ENABLE_MODULES": "YES",
      "CODE_SIGN_ENTITLEMENTS": "expovanillasdk50/expovanillasdk50.entitlements",
      "CURRENT_PROJECT_VERSION": 1,
      "ENABLE_BITCODE": "NO",
      "GCC_PREPROCESSOR_DEFINITIONS": [
        ""$(inherited)"",
        ""FB_SONARKIT_ENABLED=1"",
      ],
      "INFOPLIST_FILE": "expovanillasdk50/Info.plist",
      "IPHONEOS_DEPLOYMENT_TARGET": "13.4",
      "LD_RUNPATH_SEARCH_PATHS": ""$(inherited) @executable_path/Frameworks"",
      "MARKETING_VERSION": "1.0",
      "OTHER_LDFLAGS": [
        ""$(inherited)"",
        ""-ObjC"",
        ""-lc++"",
      ],
      "PRODUCT_BUNDLE_IDENTIFIER": ""com.killerchip.expo-vanilla-sdk50"",
      "PRODUCT_NAME": ""expovanillasdk50"",
      "SWIFT_OBJC_BRIDGING_HEADER": "expovanillasdk50/expovanillasdk50-Bridging-Header.h",
      "SWIFT_OPTIMIZATION_LEVEL": ""-Onone"",
      "SWIFT_VERSION": "5.0",
      "TARGETED_DEVICE_FAMILY": ""1,2"",
      "VERSIONING_SYSTEM": ""apple-generic"",
    },
    "isa": "XCBuildConfiguration",
    "name": "Debug",
  },
  "13B07F941A680F5B00A75B9A_comment": "Debug",
  "13B07F951A680F5B00A75B9A": {
    "baseConfigurationReference": "7A4D352CD337FB3A3BF06240",
    "baseConfigurationReference_comment": "Pods-expovanillasdk50.release.xcconfig",
    "buildSettings": {
      ""EXCLUDED_ARCHS[sdk=iphonesimulator*]"": ""arm64"",
      "ASSETCATALOG_COMPILER_APPICON_NAME": "AppIcon",
      "CLANG_ENABLE_MODULES": "YES",
      "CODE_SIGN_ENTITLEMENTS": "expovanillasdk50/expovanillasdk50.entitlements",
      "CURRENT_PROJECT_VERSION": 1,
      "INFOPLIST_FILE": "expovanillasdk50/Info.plist",
      "IPHONEOS_DEPLOYMENT_TARGET": "13.4",
      "LD_RUNPATH_SEARCH_PATHS": ""$(inherited) @executable_path/Frameworks"",
      "MARKETING_VERSION": "1.0",
      "OTHER_LDFLAGS": [
        ""$(inherited)"",
        ""-ObjC"",
        ""-lc++"",
      ],
      "PRODUCT_BUNDLE_IDENTIFIER": ""com.killerchip.expo-vanilla-sdk50"",
      "PRODUCT_NAME": ""expovanillasdk50"",
      "SWIFT_OBJC_BRIDGING_HEADER": "expovanillasdk50/expovanillasdk50-Bridging-Header.h",
      "SWIFT_VERSION": "5.0",
      "TARGETED_DEVICE_FAMILY": ""1,2"",
      "VERSIONING_SYSTEM": ""apple-generic"",
    },
    "isa": "XCBuildConfiguration",
    "name": "Release",
  },
  "13B07F951A680F5B00A75B9A_comment": "Release",
  "83CBBA201A601CBA00E9B192": {
    "buildSettings": {
      ""CODE_SIGN_IDENTITY[sdk=iphoneos*]"": ""iPhone Developer"",
      "ALWAYS_SEARCH_USER_PATHS": "NO",
      "CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED": "YES",
      "CLANG_CXX_LANGUAGE_STANDARD": ""c++20"",
      "CLANG_CXX_LIBRARY": ""libc++"",
      "CLANG_ENABLE_MODULES": "YES",
      "CLANG_ENABLE_OBJC_ARC": "YES",
      "CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING": "YES",
      "CLANG_WARN_BOOL_CONVERSION": "YES",
      "CLANG_WARN_COMMA": "YES",
      "CLANG_WARN_CONSTANT_CONVERSION": "YES",
      "CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS": "YES",
      "CLANG_WARN_DIRECT_OBJC_ISA_USAGE": "YES_ERROR",
      "CLANG_WARN_EMPTY_BODY": "YES",
      "CLANG_WARN_ENUM_CONVERSION": "YES",
      "CLANG_WARN_INFINITE_RECURSION": "YES",
      "CLANG_WARN_INT_CONVERSION": "YES",
      "CLANG_WARN_NON_LITERAL_NULL_CONVERSION": "YES",
      "CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF": "YES",
      "CLANG_WARN_OBJC_LITERAL_CONVERSION": "YES",
      "CLANG_WARN_OBJC_ROOT_CLASS": "YES_ERROR",
      "CLANG_WARN_RANGE_LOOP_ANALYSIS": "YES",
      "CLANG_WARN_STRICT_PROTOTYPES": "YES",
      "CLANG_WARN_SUSPICIOUS_MOVE": "YES",
      "CLANG_WARN_UNREACHABLE_CODE": "YES",
      "CLANG_WARN__DUPLICATE_METHOD_MATCH": "YES",
      "COPY_PHASE_STRIP": "NO",
      "ENABLE_STRICT_OBJC_MSGSEND": "YES",
      "ENABLE_TESTABILITY": "YES",
      "GCC_C_LANGUAGE_STANDARD": "gnu99",
      "GCC_DYNAMIC_NO_PIC": "NO",
      "GCC_NO_COMMON_BLOCKS": "YES",
      "GCC_OPTIMIZATION_LEVEL": 0,
      "GCC_PREPROCESSOR_DEFINITIONS": [
        ""DEBUG=1"",
        ""$(inherited)"",
      ],
      "GCC_SYMBOLS_PRIVATE_EXTERN": "NO",
      "GCC_WARN_64_TO_32_BIT_CONVERSION": "YES",
      "GCC_WARN_ABOUT_RETURN_TYPE": "YES_ERROR",
      "GCC_WARN_UNDECLARED_SELECTOR": "YES",
      "GCC_WARN_UNINITIALIZED_AUTOS": "YES_AGGRESSIVE",
      "GCC_WARN_UNUSED_FUNCTION": "YES",
      "GCC_WARN_UNUSED_VARIABLE": "YES",
      "IPHONEOS_DEPLOYMENT_TARGET": "13.4",
      "LD_RUNPATH_SEARCH_PATHS": ""/usr/lib/swift $(inherited)"",
      "LIBRARY_SEARCH_PATHS": ""\\"$(inherited)\\""",
      "MTL_ENABLE_DEBUG_INFO": "YES",
      "ONLY_ACTIVE_ARCH": "YES",
      "SDKROOT": "iphoneos",
    },
    "isa": "XCBuildConfiguration",
    "name": "Debug",
  },
  "83CBBA201A601CBA00E9B192_comment": "Debug",
  "83CBBA211A601CBA00E9B192": {
    "buildSettings": {
      ""CODE_SIGN_IDENTITY[sdk=iphoneos*]"": ""iPhone Developer"",
      "ALWAYS_SEARCH_USER_PATHS": "NO",
      "CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED": "YES",
      "CLANG_CXX_LANGUAGE_STANDARD": ""c++20"",
      "CLANG_CXX_LIBRARY": ""libc++"",
      "CLANG_ENABLE_MODULES": "YES",
      "CLANG_ENABLE_OBJC_ARC": "YES",
      "CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING": "YES",
      "CLANG_WARN_BOOL_CONVERSION": "YES",
      "CLANG_WARN_COMMA": "YES",
      "CLANG_WARN_CONSTANT_CONVERSION": "YES",
      "CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS": "YES",
      "CLANG_WARN_DIRECT_OBJC_ISA_USAGE": "YES_ERROR",
      "CLANG_WARN_EMPTY_BODY": "YES",
      "CLANG_WARN_ENUM_CONVERSION": "YES",
      "CLANG_WARN_INFINITE_RECURSION": "YES",
      "CLANG_WARN_INT_CONVERSION": "YES",
      "CLANG_WARN_NON_LITERAL_NULL_CONVERSION": "YES",
      "CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF": "YES",
      "CLANG_WARN_OBJC_LITERAL_CONVERSION": "YES",
      "CLANG_WARN_OBJC_ROOT_CLASS": "YES_ERROR",
      "CLANG_WARN_RANGE_LOOP_ANALYSIS": "YES",
      "CLANG_WARN_STRICT_PROTOTYPES": "YES",
      "CLANG_WARN_SUSPICIOUS_MOVE": "YES",
      "CLANG_WARN_UNREACHABLE_CODE": "YES",
      "CLANG_WARN__DUPLICATE_METHOD_MATCH": "YES",
      "COPY_PHASE_STRIP": "YES",
      "ENABLE_NS_ASSERTIONS": "NO",
      "ENABLE_STRICT_OBJC_MSGSEND": "YES",
      "GCC_C_LANGUAGE_STANDARD": "gnu99",
      "GCC_NO_COMMON_BLOCKS": "YES",
      "GCC_WARN_64_TO_32_BIT_CONVERSION": "YES",
      "GCC_WARN_ABOUT_RETURN_TYPE": "YES_ERROR",
      "GCC_WARN_UNDECLARED_SELECTOR": "YES",
      "GCC_WARN_UNINITIALIZED_AUTOS": "YES_AGGRESSIVE",
      "GCC_WARN_UNUSED_FUNCTION": "YES",
      "GCC_WARN_UNUSED_VARIABLE": "YES",
      "IPHONEOS_DEPLOYMENT_TARGET": "13.4",
      "LD_RUNPATH_SEARCH_PATHS": ""/usr/lib/swift $(inherited)"",
      "LIBRARY_SEARCH_PATHS": ""\\"$(inherited)\\""",
      "MTL_ENABLE_DEBUG_INFO": "NO",
      "SDKROOT": "iphoneos",
      "VALIDATE_PRODUCT": "YES",
    },
    "isa": "XCBuildConfiguration",
    "name": "Release",
  },
  "83CBBA211A601CBA00E9B192_comment": "Release",
}
`;

exports[`iOS plugin should be able to update the XCode project with excluded architectures 1`] = `
{
  "13B07F941A680F5B00A75B9A": {
    "baseConfigurationReference": "6C2E3173556A471DD304B334",
    "baseConfigurationReference_comment": "Pods-expovanillasdk50.debug.xcconfig",
    "buildSettings": {
      ""EXCLUDED_ARCHS[sdk=iphonesimulator*]"": ""arm64"",
      "ASSETCATALOG_COMPILER_APPICON_NAME": "AppIcon",
      "CLANG_ENABLE_MODULES": "YES",
      "CODE_SIGN_ENTITLEMENTS": "expovanillasdk50/expovanillasdk50.entitlements",
      "CURRENT_PROJECT_VERSION": 1,
      "ENABLE_BITCODE": "NO",
      "GCC_PREPROCESSOR_DEFINITIONS": [
        ""$(inherited)"",
        ""FB_SONARKIT_ENABLED=1"",
      ],
      "INFOPLIST_FILE": "expovanillasdk50/Info.plist",
      "IPHONEOS_DEPLOYMENT_TARGET": "13.4",
      "LD_RUNPATH_SEARCH_PATHS": ""$(inherited) @executable_path/Frameworks"",
      "MARKETING_VERSION": "1.0",
      "OTHER_LDFLAGS": [
        ""$(inherited)"",
        ""-ObjC"",
        ""-lc++"",
      ],
      "PRODUCT_BUNDLE_IDENTIFIER": ""com.killerchip.expo-vanilla-sdk50"",
      "PRODUCT_NAME": ""expovanillasdk50"",
      "SWIFT_OBJC_BRIDGING_HEADER": "expovanillasdk50/expovanillasdk50-Bridging-Header.h",
      "SWIFT_OPTIMIZATION_LEVEL": ""-Onone"",
      "SWIFT_VERSION": "5.0",
      "TARGETED_DEVICE_FAMILY": ""1,2"",
      "VERSIONING_SYSTEM": ""apple-generic"",
    },
    "isa": "XCBuildConfiguration",
    "name": "Debug",
  },
  "13B07F941A680F5B00A75B9A_comment": "Debug",
  "13B07F951A680F5B00A75B9A": {
    "baseConfigurationReference": "7A4D352CD337FB3A3BF06240",
    "baseConfigurationReference_comment": "Pods-expovanillasdk50.release.xcconfig",
    "buildSettings": {
      ""EXCLUDED_ARCHS[sdk=iphonesimulator*]"": ""arm64"",
      "ASSETCATALOG_COMPILER_APPICON_NAME": "AppIcon",
      "CLANG_ENABLE_MODULES": "YES",
      "CODE_SIGN_ENTITLEMENTS": "expovanillasdk50/expovanillasdk50.entitlements",
      "CURRENT_PROJECT_VERSION": 1,
      "INFOPLIST_FILE": "expovanillasdk50/Info.plist",
      "IPHONEOS_DEPLOYMENT_TARGET": "13.4",
      "LD_RUNPATH_SEARCH_PATHS": ""$(inherited) @executable_path/Frameworks"",
      "MARKETING_VERSION": "1.0",
      "OTHER_LDFLAGS": [
        ""$(inherited)"",
        ""-ObjC"",
        ""-lc++"",
      ],
      "PRODUCT_BUNDLE_IDENTIFIER": ""com.killerchip.expo-vanilla-sdk50"",
      "PRODUCT_NAME": ""expovanillasdk50"",
      "SWIFT_OBJC_BRIDGING_HEADER": "expovanillasdk50/expovanillasdk50-Bridging-Header.h",
      "SWIFT_VERSION": "5.0",
      "TARGETED_DEVICE_FAMILY": ""1,2"",
      "VERSIONING_SYSTEM": ""apple-generic"",
    },
    "isa": "XCBuildConfiguration",
    "name": "Release",
  },
  "13B07F951A680F5B00A75B9A_comment": "Release",
  "83CBBA201A601CBA00E9B192": {
    "buildSettings": {
      ""CODE_SIGN_IDENTITY[sdk=iphoneos*]"": ""iPhone Developer"",
      "ALWAYS_SEARCH_USER_PATHS": "NO",
      "CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED": "YES",
      "CLANG_CXX_LANGUAGE_STANDARD": ""c++20"",
      "CLANG_CXX_LIBRARY": ""libc++"",
      "CLANG_ENABLE_MODULES": "YES",
      "CLANG_ENABLE_OBJC_ARC": "YES",
      "CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING": "YES",
      "CLANG_WARN_BOOL_CONVERSION": "YES",
      "CLANG_WARN_COMMA": "YES",
      "CLANG_WARN_CONSTANT_CONVERSION": "YES",
      "CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS": "YES",
      "CLANG_WARN_DIRECT_OBJC_ISA_USAGE": "YES_ERROR",
      "CLANG_WARN_EMPTY_BODY": "YES",
      "CLANG_WARN_ENUM_CONVERSION": "YES",
      "CLANG_WARN_INFINITE_RECURSION": "YES",
      "CLANG_WARN_INT_CONVERSION": "YES",
      "CLANG_WARN_NON_LITERAL_NULL_CONVERSION": "YES",
      "CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF": "YES",
      "CLANG_WARN_OBJC_LITERAL_CONVERSION": "YES",
      "CLANG_WARN_OBJC_ROOT_CLASS": "YES_ERROR",
      "CLANG_WARN_RANGE_LOOP_ANALYSIS": "YES",
      "CLANG_WARN_STRICT_PROTOTYPES": "YES",
      "CLANG_WARN_SUSPICIOUS_MOVE": "YES",
      "CLANG_WARN_UNREACHABLE_CODE": "YES",
      "CLANG_WARN__DUPLICATE_METHOD_MATCH": "YES",
      "COPY_PHASE_STRIP": "NO",
      "ENABLE_STRICT_OBJC_MSGSEND": "YES",
      "ENABLE_TESTABILITY": "YES",
      "GCC_C_LANGUAGE_STANDARD": "gnu99",
      "GCC_DYNAMIC_NO_PIC": "NO",
      "GCC_NO_COMMON_BLOCKS": "YES",
      "GCC_OPTIMIZATION_LEVEL": 0,
      "GCC_PREPROCESSOR_DEFINITIONS": [
        ""DEBUG=1"",
        ""$(inherited)"",
      ],
      "GCC_SYMBOLS_PRIVATE_EXTERN": "NO",
      "GCC_WARN_64_TO_32_BIT_CONVERSION": "YES",
      "GCC_WARN_ABOUT_RETURN_TYPE": "YES_ERROR",
      "GCC_WARN_UNDECLARED_SELECTOR": "YES",
      "GCC_WARN_UNINITIALIZED_AUTOS": "YES_AGGRESSIVE",
      "GCC_WARN_UNUSED_FUNCTION": "YES",
      "GCC_WARN_UNUSED_VARIABLE": "YES",
      "IPHONEOS_DEPLOYMENT_TARGET": "13.4",
      "LD_RUNPATH_SEARCH_PATHS": ""/usr/lib/swift $(inherited)"",
      "LIBRARY_SEARCH_PATHS": ""\\"$(inherited)\\""",
      "MTL_ENABLE_DEBUG_INFO": "YES",
      "ONLY_ACTIVE_ARCH": "YES",
      "SDKROOT": "iphoneos",
    },
    "isa": "XCBuildConfiguration",
    "name": "Debug",
  },
  "83CBBA201A601CBA00E9B192_comment": "Debug",
  "83CBBA211A601CBA00E9B192": {
    "buildSettings": {
      ""CODE_SIGN_IDENTITY[sdk=iphoneos*]"": ""iPhone Developer"",
      "ALWAYS_SEARCH_USER_PATHS": "NO",
      "CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED": "YES",
      "CLANG_CXX_LANGUAGE_STANDARD": ""c++20"",
      "CLANG_CXX_LIBRARY": ""libc++"",
      "CLANG_ENABLE_MODULES": "YES",
      "CLANG_ENABLE_OBJC_ARC": "YES",
      "CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING": "YES",
      "CLANG_WARN_BOOL_CONVERSION": "YES",
      "CLANG_WARN_COMMA": "YES",
      "CLANG_WARN_CONSTANT_CONVERSION": "YES",
      "CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS": "YES",
      "CLANG_WARN_DIRECT_OBJC_ISA_USAGE": "YES_ERROR",
      "CLANG_WARN_EMPTY_BODY": "YES",
      "CLANG_WARN_ENUM_CONVERSION": "YES",
      "CLANG_WARN_INFINITE_RECURSION": "YES",
      "CLANG_WARN_INT_CONVERSION": "YES",
      "CLANG_WARN_NON_LITERAL_NULL_CONVERSION": "YES",
      "CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF": "YES",
      "CLANG_WARN_OBJC_LITERAL_CONVERSION": "YES",
      "CLANG_WARN_OBJC_ROOT_CLASS": "YES_ERROR",
      "CLANG_WARN_RANGE_LOOP_ANALYSIS": "YES",
      "CLANG_WARN_STRICT_PROTOTYPES": "YES",
      "CLANG_WARN_SUSPICIOUS_MOVE": "YES",
      "CLANG_WARN_UNREACHABLE_CODE": "YES",
      "CLANG_WARN__DUPLICATE_METHOD_MATCH": "YES",
      "COPY_PHASE_STRIP": "YES",
      "ENABLE_NS_ASSERTIONS": "NO",
      "ENABLE_STRICT_OBJC_MSGSEND": "YES",
      "GCC_C_LANGUAGE_STANDARD": "gnu99",
      "GCC_NO_COMMON_BLOCKS": "YES",
      "GCC_WARN_64_TO_32_BIT_CONVERSION": "YES",
      "GCC_WARN_ABOUT_RETURN_TYPE": "YES_ERROR",
      "GCC_WARN_UNDECLARED_SELECTOR": "YES",
      "GCC_WARN_UNINITIALIZED_AUTOS": "YES_AGGRESSIVE",
      "GCC_WARN_UNUSED_FUNCTION": "YES",
      "GCC_WARN_UNUSED_VARIABLE": "YES",
      "IPHONEOS_DEPLOYMENT_TARGET": "13.4",
      "LD_RUNPATH_SEARCH_PATHS": ""/usr/lib/swift $(inherited)"",
      "LIBRARY_SEARCH_PATHS": ""\\"$(inherited)\\""",
      "MTL_ENABLE_DEBUG_INFO": "NO",
      "SDKROOT": "iphoneos",
      "VALIDATE_PRODUCT": "YES",
    },
    "isa": "XCBuildConfiguration",
    "name": "Release",
  },
  "83CBBA211A601CBA00E9B192_comment": "Release",
}
`;

exports[`iOS plugin should update the Podfile with simdjson 1`] = `
"require File.join(File.dirname(\`node --print "require.resolve('expo/package.json')"\`), "scripts/autolinking")
require File.join(File.dirname(\`node --print "require.resolve('react-native/package.json')"\`), "scripts/react_native_pods")
require File.join(File.dirname(\`node --print "require.resolve('@react-native-community/cli-platform-ios/package.json')"\`), "native_modules")

platform :ios, '12.0'

require 'json'
podfile_properties = JSON.parse(File.read('./Podfile.properties.json')) rescue {}

target 'HelloWorld' do
  use_expo_modules!
  config = use_native_modules!

  use_react_native!(
    :path => config[:reactNativePath],
    :hermes_enabled => podfile_properties['expo.jsEngine'] == 'hermes'
  )

  # Uncomment to opt-in to using Flipper
  #
  # if !ENV['CI']
  #   use_flipper!('Flipper' => '0.75.1', 'Flipper-Folly' => '2.5.3', 'Flipper-RSocket' => '1.3.1')
  # end

  

  pod 'simdjson', path: File.join(File.dirname(\`node --print "require.resolve('@nozbe/simdjson/package.json')"\`)), :modular_headers => true 

  post_install do |installer|
    react_native_post_install(installer)

    # Workaround \`Cycle inside FBReactNativeSpec\` error for react-native 0.64
    # Reference: https://github.com/software-mansion/react-native-screens/issues/842#issuecomment-812543933
    installer.pods_project.targets.each do |target|
      if (target.name&.eql?('FBReactNativeSpec'))
        target.build_phases.each do |build_phase|
          if (build_phase.respond_to?(:name) && build_phase.name.eql?('[CP-User] Generate Specs'))
            target.build_phases.move(build_phase, 0)
          end
        end
      end
    end
  end

  post_integrate do |installer|
    begin
      expo_patch_react_imports!(installer)
    rescue => e
      Pod::UI.warn e
    end
  end

end
"
`;

exports[`iOS plugin should update the Podfile with simdjson only once on multiple prebuilds 1`] = `
"require File.join(File.dirname(\`node --print "require.resolve('expo/package.json')"\`), "scripts/autolinking")
require File.join(File.dirname(\`node --print "require.resolve('react-native/package.json')"\`), "scripts/react_native_pods")
require File.join(File.dirname(\`node --print "require.resolve('@react-native-community/cli-platform-ios/package.json')"\`), "native_modules")

platform :ios, '12.0'

require 'json'
podfile_properties = JSON.parse(File.read('./Podfile.properties.json')) rescue {}

target 'HelloWorld' do
  use_expo_modules!
  config = use_native_modules!

  use_react_native!(
    :path => config[:reactNativePath],
    :hermes_enabled => podfile_properties['expo.jsEngine'] == 'hermes'
  )

  # Uncomment to opt-in to using Flipper
  #
  # if !ENV['CI']
  #   use_flipper!('Flipper' => '0.75.1', 'Flipper-Folly' => '2.5.3', 'Flipper-RSocket' => '1.3.1')
  # end

  

  pod 'simdjson', path: File.join(File.dirname(\`node --print "require.resolve('@nozbe/simdjson/package.json')"\`)), :modular_headers => true 

  post_install do |installer|
    react_native_post_install(installer)

    # Workaround \`Cycle inside FBReactNativeSpec\` error for react-native 0.64
    # Reference: https://github.com/software-mansion/react-native-screens/issues/842#issuecomment-812543933
    installer.pods_project.targets.each do |target|
      if (target.name&.eql?('FBReactNativeSpec'))
        target.build_phases.each do |build_phase|
          if (build_phase.respond_to?(:name) && build_phase.name.eql?('[CP-User] Generate Specs'))
            target.build_phases.move(build_phase, 0)
          end
        end
      end
    end
  end

  post_integrate do |installer|
    begin
      expo_patch_react_imports!(installer)
    rescue => e
      Pod::UI.warn e
    end
  end

end
"
`;
